### ISSUE TITLE
Check sets for duplicate entries

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#5880]
### Current problem

Pylint check dictionaries for duplicates, but not sets.

```python
DICTIONARY = {
    1: 2,
    2: 3,
    1: 2,  # duplicate flagged
}

SET = {
    1,
    2,
    1,  # duplicate not flagged
}
```

### Desired solution

Flag duplicate set entries just as with dictionaries.

**Comment by ksaketou** [comment]
Hello, I'd like to work on that. If I understood correctly, the message we are talking about here is `duplicate-key`?

**Comment by DanielNoord** [comment]
Yes! We might also need to update the description of the message to not only mention dictionaries any more 😉

**Comment by ksaketou** [comment]
Okayy, so we don't have to create a new message for the sets?

**Comment by DanielNoord** [comment]
Hm, now that I think about it, perhaps we should create `duplicate-values`? The original [pep](https://peps.python.org/pep-0218/) refers to the entries in a set as `values`. Using `keys` might be a bit confusing? What do others think?

**Comment by Pierre-Sassoulas** [comment]
I think creating a new message ``duplicate-values`` make sense.

**Comment by nickdrozd** [comment]
The PEP says:

> Programmers are often told that they can implement sets as dictionaries with “don’t care” values. Items can be added to these “sets” by assigning the “don’t care” value to them; membership can be tested using dict.has_key...

From this perspective, set entries are more like "keys". But then the rest of the PEP refers to them as "values". And the docstring for `set` mentions "elements", as is usually done in math.

So in the interest of covering every perspective, I suggest we use `duplicate-kvalemuys`.

**Comment by ksaketou** [comment]
Hello, I am trying to run the tests but I get a `ModuleNotFoundError` at the primer tests for this line:
https://github.com/PyCQA/pylint/blob/8e9a44727545b09ccc0161d72b8b66bc92d95d88/pylint/testutils/primer.py#L5
 Any idea why this is happening?

**Comment by Pierre-Sassoulas** [comment]
Hello @ksaketou, you need to install pylint locally with ``pip3 install -e .[testutil]`` we made this package an extra in #5486 in order to not add a runtime dependency for standard users.

**Comment by ksaketou** [comment]
Hello again, I'm getting an unexpected `syntax-error` at the `symlink_module1` functional test, but I don't know how my changes could be related to this specific error message. I believe this is the line with the problem: 
https://github.com/PyCQA/pylint/blob/975550582494f9e1c0eb43336943c716120bc4e6/tests/functional/s/symlink/_binding/symlink_module.py#L1

**Comment by DanielNoord** [comment]
How did you clone the repository? That functional test is dependent on symlinks so I think it is important that the repository is cloned via `git` instead of installed via `pip` or downloaded directly from Github.

**Comment by ksaketou** [comment]
I cloned it with `git clone`. The path of this line is correct though, it's not that I miss any files.

**Comment by DanielNoord** [comment]
Did you get a output log for the syntax error_

**Comment by ksaketou** [comment]
Keeping the symlink as a string like below seems to work for my case so far. However, I don't know if there is any problem with changing this file.  

![image](https://user-images.githubusercontent.com/56515303/158571248-d6ac2c93-ce9f-43e7-bb40-fe48646db326.png)

**Comment by DanielNoord** [comment]
Yeah, if the file gets changed then the test won't work anymore. Perhaps you can revert the change and see if you can commit without any changes to the file?

Were you unable to allow `git` to create symlinks?

**Comment by ksaketou** [comment]
When I run the `symlink_module1` test with that change it passes though.  

I changed the `core.symlinks` of the .gitconfig file to True but that didn't seem to work. If I don't find anything else I will try to commit without that change as you said.

**Comment by DanielNoord** [comment]
I'll need to see the actual commit to see if nothing actually changes. This is difficult to interpret without seeing the actual changes `git` picks up on. Luckily we have the CI which should also pick up on this 😄 

I'd say go ahead with working on this, we should be able to resolve any issues with this particular test if they arise!
